Task 3/Task 3.2/Task 3.2/Task 3.2/DynamicArray.cs

```diff
@@ -0,0 +1,250 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Task_3._2
+{
+    public class DynamicArray<T> : IEnumerable<T>, ICloneable
+    {
+        private protected T[] _myArray { get; set; }
+
+        public int capacity { get; private set; } = 8;
+
+        public int length { get; private set; } = 0;
+
+        public DynamicArray()
+        {
+            _myArray = new T[capacity];
+        }
+
+        public DynamicArray(int value)
+        {
+            if (value > capacity)
+            {
+                while (capacity < value)
+                {
+                    capacity *= 2;
+                }
+            }
```

Comment:
What if I really need an array with capacity less than 8 (for example I know that it's length will not be larger than 5)

Here you can set the exact capacity